GliaWidgets/View/Chat/ChatView.swift

```diff
@@ -185,6 +187,24 @@ class ChatView: EngagementView {
         )
     }
 
+    private func addTapGesture() {
+        let tapGesture = UITapGestureRecognizer(
+            target: self,
+            action: #selector(dismissKeyboard)
+        )
+
+        tapGesture.cancelsTouchesInView = false
+
+        addGestureRecognizer(tapGesture)
+    }
+    
+    @objc
+    private func dismissKeyboard(sender: UITapGestureRecognizer) {
+        if sender.state == .ended {
```

Comment:
do we need to handle not only `ended` but `cancelled` and `failed` cases also?

Reply:
Not really. Since UITapGesture is discrete gesture, it really doesn't trigger for any other state.
Considering that, there is really no need to specifically check for end state either, however for some reason (maybe future-proofing?) Apple's documentation example suggest we check the state: https://developer.apple.com/documentation/uikit/uitapgesturerecognizer